### PR TITLE
Rework game resolution management (part 1)

### LINKED ISSUE #114: Suggestion: Change Renderer Behaviour
Changing Resolution doesnt change the resolution of the back buffer (it is the same size as device's main resolution)
This means that even though the Resolution is set as 800x600, the Renderer still renders as 1920x1080 (my native resolution).

What the renderer does instead is a bit complex.
The vertices are snapped to the "pixels" of the resolution in the settings which results in heavily distorted models.
I noticed that the game isn't rendered in my full resolution though. The vertical resolution stays the same (1080), while the horizontal resolution changes with the resolution i set.

Here is a screenshot of original TombATI (custom resolution of 160x120, extreme case but for showcasing purposes) and reproducible with Tomb1Main:
screenshot0003

Notice how the image is 1080px high, but the "resolution" of Lara is as if it is 160x120.

I'd expect to have a smaller frame buffer (160x120 in this case), i.e. rendering what you actually set in the settings.

I also noticed that the Title menu is rendered independently. Changing resolution does not affect the tile menu

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Functional:
- removed `hi_res` option. Instead the resolution is now saved in `resolution_width/height` keys
- resolution changes are applied immediately.
- changed the desktop/user-supplied resolution to be positioned first, not last. If the user changes `resolution_width/height` to something weird, these values will override the first resolution and it will be selected.
- fixed fade to black animation for pictures ending on a non-black frame (so when Eidos logo ends fading and the main menu is about to load, we still very dark Eidos logo instead of black screen in the current release; this is fixed by this 3673ae5)

Cosmetic:
- dropped external index management in favor of direct width/height comparisons
- removed a couple of pointless functions such as `Output_InitialisePolyList` and `Output_ClearScreen`
- renamed `Output_DisplayPicture` and similar to use new term, `BackdropImage`. Changed `display` to `load` as it doesn't actually display anything.

This doesn't fix #114 but it's a start.
This is best viewed on a commit-by-commit basis.

### CODE REVIEW

**Review by walkawayy — APPROVED** [review]
Changes look nice! The instant resolution change is much better too.

I tested different resolutions and everything worked for me including screenshots.

**Comment by rr-** [comment]
I think the changelog could use improvements but IDK how to include the entire scope of the changes without being too wordy about it.

**Review by lahm86 — APPROVED** [review]
LGTM
